### ISSUE TITLE
Style/RedundantRegexpEscape incorrectly flags line continuations as redundant regexp escapes

### ISSUE THREAD

**Issue by zofrex** [rubocop-hq/rubocop#8583]
The Style/RedundantRegexpEscape cop flags line continuations – "\\" at the end of a line to indicate it continues on the next line – that occur within regexps as redundant regexp escapes, which they are not.

--------

## Expected behavior

Rubocop should happily accept a \ at the end of a line in a regexp. It is not redundant, it is semantically meaningful, and removing it changes the meaning of the regexp.

## Actual behavior

The \ is flagged as redundant:

```
test/test.rb:72:48: C: Style/RedundantRegexpEscape: Redundant escape inside regexp literal
          foo = /The quick brown fox jumps over\ ...
```

## Steps to reproduce the problem

Running Rubocop with the Style/RedundantRegexpEscape cop enabled over code like this:

```
          foo = /The quick brown fox jumps over\
the lazy dog/
```

## RuboCop version

```
% bundle exec rubocop -V
0.89.1 (using Parser 2.7.1.4, rubocop-ast 0.3.0, running on ruby 2.7.1 x86_64-darwin19)
```

**Comment by bbatsov** [comment]
Yeah, that's a bug, but in general I think you'd do well to use the `x` modifier instead https://rubystyle.guide/#comment-regexes
